src/App.tsx

```diff
@@ -1,26 +1,173 @@
+import { TodoFooter } from './components/ToDoFooter';
+import { Error } from './components/Error';
+import { type Todo } from './types/Todo';
+import { createTodo } from '../src/components/api/todos';
+import { FilterButtons } from './types/FilterType';
+import { Errors } from './types/EnumedErrors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors>(Errors.NoLetters);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [tempIds, setTempIds] = useState<number[]>([]);
+  const [newToDoTitle, setNewToDoTitle] = useState('');
+  const [filterButton, setFilterButton] = useState<FilterButtons>(
+    FilterButtons.All,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleError = (errorsTexts: Errors) => {
+    setError(errorsTexts);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError(Errors.UnableToLoad);
+        setTimeout(() => {
+          setError(Errors.NoLetters);
+        }, 3000);
+      });
+  }, [error]);
+
+  const handleAddingTodos = (event: React.FormEvent<Element>) => {
+    event.preventDefault();
+    if (!newToDoTitle.trim()) {
+      setError(Errors.TitleEmpty);
+      setTimeout(() => {
+        setError(Errors.NoLetters);
+      }, 3000);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    const createdTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: newToDoTitle.trim(),
+      completed: false,
+      editted: false,
+    };
+
+    setTemporaryTodo(createdTodo);
+    setTempIds(currentIds => [...currentIds, 0]);
+
+    return createTodo(createdTodo)
+      .then(curTodo => {
+        setTodos([...todos, curTodo]);
+        setNewToDoTitle('');
+      })
+      .catch(() => {
+        setError(Errors.UnableToAdd);
+        setTimeout(() => {
+          setError(Errors.NoLetters);
+        }, 3000);
+      })
+      .finally(() => {
+        setTemporaryTodo(null);
+        setTempIds([]);
+        setIsLoading(false);
+      });
+  };
+
+  const handleDeletedTodo = (idNumber: number) => {
+    setIsLoading(true);
+    setTempIds(currentIds => [...currentIds, idNumber]);
+
+    return deleteTodo(idNumber)
+      .then(() => {
+        setTodos(currTodos => currTodos.filter(todo => todo.id !== idNumber));
+      })
+      .catch(() => {
+        setError(Errors.UnableToDelete);
+        setTimeout(() => {
+          setError(Errors.NoLetters);
+        }, 3000);
+      })
+      .finally(() => {
+        setTempIds([]);
+        setIsLoading(false);
+      });
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const deleteOnlyCompleted = () => {
+    completedTodos.forEach(todo => handleDeletedTodo(todo.id));
+  };
+
+  const allTodosAreCompleted =
+    todos.filter(todo => todo.completed).length === todos.length;
```

Comment:
use every method and change name to areAllTodosCompleted to indicate that this variable is boolean